analyzer/block/block.go

```diff
@@ -259,6 +266,16 @@ func (b *blockBasedAnalyzer) ensureSlowSyncPrerequisites(ctx context.Context) (o
 	return true
 }
 
+func (b *blockBasedAnalyzer) sendQueueLengthMetric(ctx context.Context, height uint64) error {
+	queueLength, err := b.processor.QueueLength(ctx, height)
+	if err != nil {
+		b.logger.Warn("error fetching queue length for %s: %w", b.analyzerName, err)
```

Comment:
Nit: This method (`sendQueueLengthMetric()`) is only called once, and the caller already logs any errors. So let's log either here or there. I'd say it's more canonical to log outside, and return a `fmt.Errorf()` here.